tests/test_ingestion.py

```diff
@@ -43,127 +45,184 @@ def setUpClass(cls):
-        self._calibRepo = os.path.join(self._repo, 'calibs')
-        templateRepo = os.path.join(IngestionTestSuite.testApVerifyData, 'repoTemplate')
-
-        # Initialize as a valid repository
-        # Can't call copytree on (templateRepo, self._repo) because latter already exists
-        testFiles = os.listdir(templateRepo)
-        for testFile in testFiles:
-            original = os.path.join(templateRepo, testFile)
-            copy = os.path.join(self._repo, testFile)
-            if os.path.isdir(original):
-                shutil.copytree(original, copy)
+        # Mandatory argument to _doIngest*, used by _doIngestDefects to unpack tar
+        self._repo = self._calibRepo = tempfile.mkdtemp()
+        self.addCleanup(shutil.rmtree, self._repo, ignore_errors=True)
+
+        def mockGet(datasetType, dataId=None):
+            """Minimally fake a butler.get().
+            """
+            if "raw_filename" in datasetType:
+                matchingFiles = [datum['file'] for datum in IngestionTestSuite.rawData
+                                 if datum['visit'] == dataId['visit']]
+                return [os.path.join(self._repo, file) for file in matchingFiles]
+            elif "bias_filename" in datasetType:
+                matchingFiles = [datum['file'] for datum in IngestionTestSuite.calibData
+                                 if datum['type'] == 'bias']
+                return [os.path.join(self._repo, file) for file in matchingFiles]
+            elif "flat_filename" in datasetType:
+                matchingFiles = [datum['file'] for datum in IngestionTestSuite.calibData
+                                 if datum['type'] == 'flat' and datum['filter'] == dataId['filter']]
+                return [os.path.join(self._repo, file) for file in matchingFiles]
             else:
-                shutil.copy2(original, copy)
+                return None
 
-        # Initialize calib repo
-        # Making the directory appears to be both necessary and sufficient
-        os.mkdir(self._calibRepo)
+        patcherButler = unittest.mock.patch("lsst.daf.persistence.Butler", autospec=True)
+        self._butler = patcherButler.start()
+        self._butler.getMapperClass.return_value = lsst.obs.test.TestMapper
+        self._butler.return_value.get = mockGet
+        self.addCleanup(patcherButler.stop)
 
-        obsDir = os.path.join(getPackageDir('obs_test'), 'config')
-        config = ingestion.DatasetIngestConfig()
-        config.load(os.path.join(obsDir, 'datasetIngest.py'))
-        self._task = ingestion.DatasetIngestTask(config=config)
+        self._task = ingestion.DatasetIngestTask(config=IngestionTestSuite.config)
+
+    def setUpRawRegistry(self):
+        """Mock up the RegisterTask used for ingesting raw data.
+
+        This method initializes ``self._registerTask`` and ``self._registryHandle``.
+
+        Behavior is undefined if more than one of `setUpRawRegistry`, `setUpCalibRegistry`,
```

Comment:
I don't think it's that important -- I mean "undefined" in the sense of "I'm not supporting that case so I don't control what will happen" rather than "your hard drive will wipe itself."